cedar-policy/src/api.rs

```diff
@@ -885,35 +885,32 @@ impl PartialResponse {
     /// Returns an over-approximation of the set of determining policies
     ///
     /// This is all policies that may be determining for any substitution of the unknowns
-    pub fn may_be_determining(&self) -> impl Iterator<Item = &PolicyId> {
-        self.0.may_be_determining().map(PolicyId::ref_cast)
+    pub fn may_be_determining(&self) -> impl Iterator<Item = Policy> + '_ {
+        self.0.may_be_determining().map(Policy::from_ast)
     }
 
     /// Returns an under-approximation of the set of determining policies
     ///
-    /// This is all policies that must be determining for all possible substitutions of the unknowns
-    pub fn must_be_determining(&self) -> impl Iterator<Item = &PolicyId> {
-        self.0.must_be_determining().map(PolicyId::ref_cast)
+    /// This is all policies that must be determining for all possible substitutions of the unknowns.
+    /// Therefor, this set will end up only containing policies whose residuals are simple `true`
+    /// statements (whether `permit` or `forbid` policies)
```

Comment:
This comment might be a little deceiving, if I was unfamiliar with things, I might read this and think it means that it will actually return all policies whose residuals are simple `true`.  It might be good to clarify explicitly that it won't necessarily return all policies whose residuals are simple `true`; and/or to use the same language here and in `definitely_satisfied()` (there, we just say "definitely satisfied" and don't refer to "simple `true`", but we could).

Reply:
Hmm yeah, I'll workshop it a little bit more. I added that text because someone reported they were confused about what `must_be` would return.

Reply:
New revision